packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -113,42 +113,44 @@ class TreeItem extends UIComponent<WithAsProp<TreeItemProps>> {
       e.preventDefault()
       e.stopPropagation()
 
-      _.invoke(this.props, 'onTitleClick', e, this.props)
+      this.handleTitleClick(e)
     },
-    collapseOrReceiveFocus: e => {
-      const { items, open } = this.props
+    receiveFocus: e => {
+      e.preventDefault()
+      e.stopPropagation()
 
+      this.itemRef.current.focus()
+    },
+    collapse: e => {
```

Comment:
We've all agreed splitting the handlers into single purpose functions is correct.  let's get this merged in a separate PR.  we'll continue to discuss the `shouldHandle` and other concerns in this PR.